### ISSUE TITLE
rustc-polonius integration 

### ISSUE THREAD

**Issue by nikomatsakis** [rust-lang-nursery/polonius#44]
Here is a plan for how to do a simple rustc-polonius integration.

- [x] create `polonius-engine`  library crate that exports `AllFacts` and associated newtypes
  - it *could* have `Output` too, but I think that can wait until later, as that sidesteps the question of where datafrog lives etc
  - but it should *not* have the interning nor the code for handling tab-separated files etc
- [x] make rustc use this crate and remove the duplicate copy of `AllFacts` that is found there
  - this will imply convering from the rustc types (like `RegionVid`) and using the ones from polonius
  - for now, I would do this at the point where we generate facts, seems fine (rather then doing it deeply through rustc)
- [x] move the `Output` and datafrog code into `polonius-engine`
  - hopefully the datafrog version we need is in crates.io by then
- [x] port rustc borrow checker to use `Output` results (when given the `-Zpolonius` flag) instead of the NLL region check
  - basically similar to how I did it in [this branch](https://github.com/nikomatsakis/rust/tree/nll-alias-analysis-with-differential-dataflow)
  - but I’d prefer to use the `errors` relations that lqd + spastorino have been adding, which should I think be fine too. When we see that some path is being modified that invalidates the loan `L` at the point `P`, instead of looking for `borrow_live_at(L, P)`, we look for `error(L, P)`. Seems fine.
